src/components/form/fields/DateField.tsx

```diff
@@ -47,7 +54,7 @@ export function DateField({
                 validationText: i18n.t('Required'),
             })
         } else {
-            setValidation(payload?.validation || { error: false })
+            setValidation(payload?.validation || { error: false, valid: true })
         }
         input.onChange(payload?.calendarDateString || '')
         input.onBlur()
```

Comment:
Cant select the line below, but I think this:
```valid={validation?.valid && input?.value !== ''}```

should also check for undefined? Right now you will get a "Check" next to the field if it's undefined.
We could also format undefined to `''` though. 

eg:

```
      format: (value) =>
            value ? value.substring(0, 10) : '',
```